cmd/postgres-operator/main.go

```diff
@@ -190,3 +190,102 @@ func isOpenshift(ctx context.Context, cfg *rest.Config) bool {
 
 	return false
 }
+
+func isGKE(ctx context.Context, cfg *rest.Config) bool {
+	log := logging.FromContext(ctx)
+
+	const groupName, kind = "cloud.google.com", "BackendConfig"
+
+	client, err := discovery.NewDiscoveryClientForConfig(cfg)
+	assertNoError(err)
+
+	groups, err := client.ServerGroups()
+	if err != nil {
```

Comment:
You don't need `if err != nil` checks, just use `assertNoError(err)`. Applies to other places as well.